### DISCUSSION TITLE
Proposal: Track action should not be bypassed if request matches safelist

### DISCUSSION THREAD

**Discussion by fenelon** [kickstarter/rack-attack#525] [Ideas & Proposals]
Would like to hear the maintainers on this. Could implement if we agree.

**Comment by grzuy** [comment]
Hi @fenelon,

Would you mind giving more context on what are you trying to accomplish with "Tracks" that you can't with how it currently works? Give more details on what change would you want to see, and how that would help you or others?

Thank you!

**Comment by fenelon** [comment]
Hey @grzuy,

For instance, 

1) I want to be able to track all bot/crawler requests by user agent and, at the same time, 
2) throttle or block especially greedy ones, but 
3) always allow specific search engines or tools
